packages/mobile/src/styles/makeStyles.ts

```diff
@@ -1,87 +1,65 @@
-import { useEffect, useMemo, useRef } from 'react'
-
-import { isEqual } from 'lodash'
-import hash from 'object-hash'
-import type { TextStyle, ViewStyle, ImageStyle } from 'react-native'
+import { Theme } from '@audius/common'
+import type { ImageStyle, TextStyle, ViewStyle } from 'react-native'
 import { StyleSheet } from 'react-native'
 
-import type { ThemeColors, Theme as ThemeType } from '../utils/theme'
-import { useThemeVariant, useThemeColors } from '../utils/theme'
+import type { ThemeColors } from 'app/utils/theme'
+import {
+  matrixTheme,
+  defaultTheme,
+  darkTheme,
+  useThemeVariant
+} from 'app/utils/theme'
 
 import { spacing } from './spacing'
 import { typography } from './typography'
 
-const useMemoCompare = <Next>(
-  next: Next,
-  compare: (a?: Next, b?: Next) => boolean
-): Next => {
-  const previousRef = useRef<Next | undefined>()
-  const previous = previousRef.current
-
-  // Pass previous and next value to compare function
-  // to determine whether to consider them equal.
-  const isEqual = compare(previous, next)
-
-  // If not equal update previousRef to next value.
-  // We only update if not equal so that this hook continues to return
-  // the same old value if compare keeps returning true.
-  useEffect(() => {
-    if (!isEqual) {
-      previousRef.current = next
-    }
-  })
-
-  // Finally, if equal then return the previous value
-  // Note: isEqual == 'true' implies previous is type Next
-  return isEqual ? (previous as Next) : next
-}
-
-type Theme = {
+type StylesOptions = {
   palette: ThemeColors
   typography: typeof typography
   spacing: typeof spacing
-  type: ThemeType
+  type: Theme
 }
 
-// TODO: This doesn't cover style props for icons
 type NamedStyles<T> = { [P in keyof T]: ViewStyle | TextStyle | ImageStyle }
 
-type Styles<T extends NamedStyles<T>, PropsT> = (
-  theme: Theme,
-  props: PropsT
-) => T | NamedStyles<T>
-
-type UseStyles<T, PropsT> = PropsT extends undefined
-  ? () => T
-  : (props: PropsT) => T
-
-const styleCache = {}
-
-export const makeStyles = <
-  PropsT = undefined,
-  T extends NamedStyles<T> = NamedStyles<any>
->(
-  styles: Styles<T, PropsT>
-): UseStyles<T, PropsT> => {
-  return ((props?: PropsT): T => {
+export const makeStyles = <T extends NamedStyles<T> | NamedStyles<any>>(
```

Comment:
My understanding is that this is the crux of the change -- we don't use hooks and instead use StyleSheet.create here, which I think our old application may have done everywhere throughout the components themselves.

Probably a good testament to revisiting our overall component structure and comparing that with some standard RN cookbook type examples to make sure we're not over engineering some other areas.

What a wonderful find you had here!! Probably couldn't have done it without Flipper -- kudos 👏

Reply:
so actually we still use hooks @raymondjacobson, its just that we dont defer the compilation till inside the hook, we do the majority of the work beforehand (ie create the stylesheets beforehand)

Reply:
Yes. all makes sense. I think maybe a side effect of trying to be fancy though? an extremely hard to maintain but very wet codebase would probably have defined all the styles statically at the top like this https://snack.expo.dev